modules/RTC/TPCUtils.js

```diff
@@ -512,17 +513,17 @@ export class TPCUtils {
             return Object.values(CodecMimeType).find(value => value === codec.toLowerCase());
         }
 
-        return defaultCodec;
+        return CodecMimeType.VP8;
     }
 
     /**
      * Returns the codecs in the current order of preference as configured on the peerconnection.
      *
-     * @param {RTCSessionDescription} - The local description to be used.
+     * @param {string} - The local SDP to be used.
      * @returns {Array}
      */
-    getConfiguredVideoCodecs(description) {
-        const currentSdp = description?.sdp ?? this.pc.localDescription?.sdp;
+    getConfiguredVideoCodecs(sdp) {
```

Comment:
I suggest you break this function into 2:

```
getConfiguredVideoCodecs(sdp) {
}

_getConfiguredVideoCodecsImpl(parsedSdp) {
}
```

With the former calling the latter, after parsing the SDP.

This will allow you to call the 2nd one in the munge code and doing it all in 1 pass!